### PR TITLE
465 automate production of diagram pdfs for all regimes in a given hydraulic scheme

### PR DESCRIPTION
**Pull request by ahobeost**

### CODE REVIEW

**Comment by ahobeost** [comment]
Inkscape path will likely need to be set for the matplotlib tests to pass

File: trnsysGUI/WTap_main.py [hunk 1]
```diff
@@ -64,7 +65,7 @@ def changeSize(self):
         return w, h
 
     def exportMassFlows(self):
-        resStr = "Mfr" + self.displayName + " = 1000" + "\n"
+        resStr = "Mfr" + self.displayName + f" = {self.massFlowRateInKgPerH}" + "\n"
```
Comment (ahobeost) [inline trnsysGUI/WTap_main.py:68]: Make format string throughout

File: trnsysGUI/pythonInterface/regimeExporter/exportRegimes.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    writeToCsv(pumpsAndValvesAndValues, regimeFileName)
+
+
+def getPumpsAndValvesWithValuesFromJson(projectJson):
+    with open(projectJson, "r", encoding="utf-8") as openFile:
+        jsonValues = json.load(openFile)
+
+    data = {}
+    undesiredBlocks = [".__BlockDct__", "IDs", "Strings"]
+    for block in jsonValues["Blocks"]:
```
Comment (ahobeost) [inline trnsysGUI/pythonInterface/regimeExporter/exportRegimes.py:17]: Get attribute names from serialization model.
Maybe depends on how this evolves in the future.

File: trnsysGUI/pythonInterface/regimeExporter/exportRegimes.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        if block not in undesiredBlocks:
+            curDict = jsonValues["Blocks"][block]
+            if "BlockName" in curDict:
+                curBlockName = curDict["BlockName"]
+                if curBlockName == "Pump":
+                    desiredValueName = "massFlowRateInKgPerH"
+                    data = getData(curDict, data, desiredValueName)
+                elif curBlockName == "TVentil":
```
Comment (ahobeost) [inline trnsysGUI/pythonInterface/regimeExporter/exportRegimes.py:25]: Serialization model of Tventil doesn't know about this yet

File: trnsysGUI/pythonInterface/regimeExporter/renderDiagramOnPDFfromPython.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            self._makeDiagramFiles()
+
+        regimeValues = _gdr.getRegimes(self.dataDir / self.regimesFileName, onlyTheseRegimes)
+        pumpsAndValvesNames = list(regimeValues.columns)
+        pumpsAndValves, mainTaps = self.getPumpsAndValvesAndMainTaps(pumpsAndValvesNames)
+
+        self._simulateAndVisualizeMassFlows(mainTaps, pumpTapPairs, pumpsAndValves, regimeValues)
+
+    def _makeDiagramFiles(self, regimeName="diagram") -> None:
+        pdfName = str(self.resultsDir) + "\\" + self.projectName + "_" + regimeName + ".pdf"
```
Comment (ahobeost) [inline trnsysGUI/pythonInterface/regimeExporter/renderDiagramOnPDFfromPython.py:52]: Always use paths, and change to string for external dependencies when needed

**Comment by ahobeost** [comment]
Check inkscape for pdf cropping

**Comment by zuckerruebe** [comment]
@ahobeost: Don't remember the status here...can we merge this?

**Pull request merged**